Upstream/renesas ra input capture channel b pwm

duynguyenxa opened the pull request:
This commit add support for input capture on GPT GTIOCnB pin for Renesas RA devices

duynguyenxa committed:
drivers: pwm: renesas: Add input capture on gtiocb

This commit add support for input capture feature for PWM GPT
on Renesas RA MCU on GTIOCB pin
This use the combination of input capture interrupt B and
overflow interrupt to get pulse and period information
This commit also add modify to all impact device tree config
using renesas,ra-pwm node

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>

khoa-nguyen-18 commented on include/zephyr/dt-bindings/pwm/renesas_ra_pwm.h:
```diff
@@ -0,0 +1,12 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_RENESAS_RZ_PWM_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_RENESAS_RZ_PWM_H_
```
```suggestion
#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_RENESAS_RA_PWM_H_
#define ZEPHYR_INCLUDE_DT_BINDINGS_RENESAS_RA_PWM_H_
```

khoa-nguyen-18 commented on boards/renesas/ek_ra4e2/ek_ra4e2.dts:
```diff
@@ -166,8 +166,8 @@
 	pinctrl-names = "default";
-	interrupts = <63 1>, <64 1>;
-	interrupt-names = "gtioca", "overflow";
+	interrupts = <63 1>, <63 1>, <64 1>;
```
duplicate `<63 1>`

khoa-nguyen-18 commented on dts/bindings/pwm/renesas,ra-pwm.yaml:
```diff
@@ -22,6 +22,22 @@ properties:
+          "gtiocb": Input capture B
+          "overflow" : Overflow interrupt
+
   interrupt-names:
     required: true
```
duplicate `interrupt-names:` property, please remove this one

khoa-nguyen-18 reviewed (commented):
Hi, I have some comments regarding typos and duplicates.

duynguyenxa committed:
drivers: pwm: Fix naming of function define to match with API

Change argument naming in function define "pin" to "channel" to
match with Zephyr PWM API declare

Signed-off-by: Duy Nguyen <duy.nguyen.xa@renesas.com>
